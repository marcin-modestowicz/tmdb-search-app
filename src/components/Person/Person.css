.root {
  box-sizing: border-box;
  margin-top: 1em;
  padding: 1em 1em;
  width: 100%;
  max-width: 40em;
  font-family: sans-serif;
}

.root h3,
.root h4 {
  margin-top: 0;
}

.content > img {
  float: left;
  margin: 0 1em 1em 0;
  width: 50%;
  max-width: 185px;
}

.content ul {
  padding: 0;
  list-style-position: inside;
}

.content li {
  line-height: 1.5em;
  margin-bottom: 0.5em;
}
