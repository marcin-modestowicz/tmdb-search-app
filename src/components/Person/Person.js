// @flow

import React, { Component } from "react";
import Image from "../Image";
import styles from "./Person.css";

type Props = {
  person: Object
};

const Person = ({ person }: Props) => {
  const { profile_path, name, known_for } = person;
  const hasProfilePhoto = profile_path != null;
  const isKnownFrom = known_for != null;

  return (
    <div className={styles.root}>
      <h3>{name}</h3>
      <div className={styles.content}>
        {hasProfilePhoto && <Image name={profile_path} />}
        {isKnownFrom && (
          <div>
            <h4>Known for:</h4>
            <ul>
              {known_for.map(movie => (
                <li key={movie.id}>{movie.title || movie.name}</li>
              ))}
            </ul>
          </div>
        )}
      </div>
    </div>
  );
};

export default Person;
