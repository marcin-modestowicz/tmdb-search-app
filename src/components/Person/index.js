// @flow
export { default } from "./Person";
