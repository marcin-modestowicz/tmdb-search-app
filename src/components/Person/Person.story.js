// @flow
import React from "react";
import { storiesOf } from "@storybook/react";
import Person from "./Person";

storiesOf("Person", module).add("simple", () => (
  <Person
    person={{
      profile_path: "/ce84udJZ9QRSR44jxwK2apM3DM8.jpg",
      name: "Sean Connery",
      known_for: [
        {
          id: 1,
          title: "Indiana Jones and the Last Crusade"
        },
        {
          id: 2,
          title: "The Rock"
        }
      ]
    }}
  />
));
