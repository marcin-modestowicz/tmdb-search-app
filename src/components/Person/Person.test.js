// @flow
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Person from "./Person";

describe("Person component", () => {
  test("should render properly", () => {
    const personData = {
      profile_path: "/ce84udJZ9QRSR44jxwK2apM3DM8.jpg",
      name: "Sean Connery",
      known_for: [
        {
          id: 1,
          title: "Indiana Jones and the Last Crusade"
        },
        {
          id: 2,
          title: "The Rock"
        }
      ]
    };
    const person = shallow(<Person person={personData} />);
    expect(toJson(person)).toMatchSnapshot();
  });
});
