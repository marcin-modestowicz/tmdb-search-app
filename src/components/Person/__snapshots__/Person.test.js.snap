// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person component should render properly 1`] = `
<div
  className="root"
>
  <h3>
    Sean Connery
  </h3>
  <div
    className="content"
  >
    <Image
      name="/ce84udJZ9QRSR44jxwK2apM3DM8.jpg"
    />
    <div>
      <h4>
        Known for:
      </h4>
      <ul>
        <li
          key="1"
        >
          Indiana Jones and the Last Crusade
        </li>
        <li
          key="2"
        >
          The Rock
        </li>
      </ul>
    </div>
  </div>
</div>
`;
